Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/TapToPlace/TapToPlace.cs

```diff
@@ -0,0 +1,467 @@
+            {
+
+                IReadOnlyList<IMixedRealitySpatialAwarenessMeshObserver> observers = spatialAwarenessSystemAcecess.GetDataProviders<IMixedRealitySpatialAwarenessMeshObserver>();
+
+                foreach (IMixedRealitySpatialAwarenessMeshObserver observer in observers)
+                {
+                    return observer.DisplayOption;
+                }
+
+                // if the spatial awareness is not enabled
+                return SpatialAwarenessMeshDisplayOptions.None;   
+            }
+        }
+
+        private IMixedRealityDataProviderAccess spatialAwarenessSystemAcecess;
+
+        [SerializeField]
+        [Tooltip("If false, the game object to place will not change its rotation according to the surface hit.  The object will" +
+            "remain facing the camera while it is in the placing state.  If true, the object will rotate according to the surface normal" +
+            "if there is a hit.")]
+        private bool rotateAccordingToSurface = false;
+
+        /// <summary>
+        /// If false, the game object to place will not change its rotation according to the surface hit.  The object will 
+        /// remain facing the camera while it is in the placing state.  If true, the object will rotate according to the surface normal
+        /// if there is a hit."
+        /// </summary>
+        public bool RotateAccordingToSurface
+        {
+            get => rotateAccordingToSurface;
+            set
+            {
+                if (rotateAccordingToSurface != value)
+                {
+                    rotateAccordingToSurface = value;
+                }
+            }
+        }
+
+        private const int IgnoreRaycastLayer = 2;
+
+        private const int DefaultLayer = 0;
+
+        // The current ray is based on the TrackedTargetType (Controller Ray, Head, Hand Joint)
+        protected RayStep currentRay;
+
+        protected RaycastHit currentHit;
+
+        protected float previousFrameNumber = 0;
+
+        protected override void Start()
+        {
+            // Solver base class
+            base.Start();
+
+            // If tap to place is added via script, set the GameObjectToPlace as this gameobject 
+            if (GameObjectToPlace == null)
+            {
+                GameObjectToPlace = gameObject;
+            }
+
+            if(!ColliderPresent)
+            {
+                Debug.LogError("The GameObjectToPlace does not have a collider attached, please attach a collider");
+            }
+
+            SurfaceNormalOffset = GetComponent<Collider>().bounds.extents.z;
+
+            // What if the spatial awareness is not enabled?
+            spatialAwarenessSystemAcecess = CoreServices.SpatialAwarenessSystem as IMixedRealityDataProviderAccess;
+
+            // Get the spatial mesh display option on start
+            // I do not know if this is legal, check on this
+            initialSpatialMeshDisplayOption = InitialSpatialMeshDisplayOption;
```

Comment:
Same here, I know this is just a draft. But just adding note as reminder to clean up here